src/__tests__/mount.spec.ts

```diff
@@ -1,73 +1,115 @@
-import { build } from "../embed";
+import { buildEmbedPlugin } from "../embed";
 import { mount } from "../mount";
 import { createNoopEmbed } from "./helpers";
 
 describe("mount", () => {
-  describe("nestedEditorProps typesafety", () => {
-    it("should provide typesafe nestedEditorProps to its consumer", () => {
-      const props = {
-        type: "richText",
-        name: "prop1",
+  describe("nodeView typesafety", () => {
+    it("should provide typesafe nodeView to its consumer", () => {
+      const fieldSpec = {
+        prop1: {
+          type: "richText",
+        },
+      } as const;
+      mount(
+        "testEmbed",
+        fieldSpec,
+        () => () => null,
+        (_, __, ___, fieldNodeViews) => {
+          // Prop1 is derived from the fieldSpec
+          fieldNodeViews.prop1;
+        },
+        () => null,
+        { prop1: "text" }
+      );
+    });
+
+    it("should not typecheck when props are not provided", () => {
+      const fieldSpec = {
+        notProp1: {
+          type: "richText",
+        },
+      } as const;
+      mount(
+        "testEmbed",
+        fieldSpec,
+        () => () => null,
+        (_, __, ___, fieldNodeViews) => {
+          // @ts-expect-error – prop1 is not available on this object,
+          // as it is not defined in `fieldSpec` passed into `mount`
+          fieldNodeViews.prop1;
+        },
+        () => null,
+        { notProp1: "text" }
+      );
+    });
+  });
+
+  describe("field typesafety", () => {
+    it("should provide typesafe fields to its consumer", () => {
+      const fieldSpec = {
+        prop1: {
+          type: "richText",
+        },
       } as const;
       mount(
         "testEmbed",
-        [props],
+        fieldSpec,
         () => () => null,
-        (_, __, ___, nestedEditorProps) => {
-          // Prop1 is derived from the props
-          nestedEditorProps.prop1;
+        (fields) => {
+          // Prop1 is derived from the fieldSpec, and is a string b/c it's a richText field
+          fields.prop1.toString();
```

Comment:
It would be good to see an equivalent test for a checkbox field. Probably should have mentioned that in #23...